runtime/src/bank.rs

```diff
@@ -837,6 +838,12 @@ pub struct Bank {
 
     /// Collected fee details
     collector_fee_details: RwLock<CollectorFeeDetails>,
+    /// A lock which execution threads will acquire the read lock during execution
+    /// until cost-tracker updates are complete.
+    /// The cost-tracking reporting will acquire the write lock before reporting,
+    /// this is to avoid cost-tracker-stats from being reported before the updates
+    /// are complete.
+    cost_tracker_report_lock: RwLock<()>,
```

Comment:
not happy with this location, but couldn't figure out a good way to get this on the cost-tracker since cost-tracker is hidden behind a RwLock. Could probably do some Arc-clone, but don't want the overhead per batch (don't want to upset @ryoqun 😉 )